Assets/Scripts/EOSLobbyManager.cs

```diff
@@ -1696,6 +1699,75 @@ private void OnRTCRoomUpdateReceivingCompleted(ref UpdateReceivingCallbackInfo d
             }
         }
 
+        /// <summary>
+        /// Blocks the Local Lobby Member from receiving/sending audio from/to a Target Lobby Member. Only affects this Local Member. Does not affect other Lobby Members.
+        /// </summary>
+        /// <param name="targetUserId">The Target Member's ProductUserID</param>
+        /// <param name="status">To block or to unblock? True to Block, False to Unblock</param>
+        /// <param name="blockRTCParticipantComplete">Callback when BlockTargetRTCParticipant is completed</param>
+        public void UpdateBlockStatusForRTCParticipant(ProductUserId targetUserId, bool status, OnLobbyCallback blockRTCParticipantComplete)
+        {
+            RTCInterface rtcHandle = EOSManager.Instance.GetEOSRTCInterface();
+            RTCAudioInterface rtcAudioHandle = rtcHandle.GetAudioInterface();
+
+            BlockParticipantOptions blockParticipantOptions = new BlockParticipantOptions()
+            {
+                LocalUserId = EOSManager.Instance.GetProductUserId(),
+                RoomName = CurrentLobby.RTCRoomName,
+                ParticipantId = targetUserId,
+                Blocked = status
+            };
+
+            rtcHandle.BlockParticipant(ref blockParticipantOptions, blockRTCParticipantComplete, OnRTCBlockParticipantCompleted);
+        }
+
+        private void OnRTCBlockParticipantCompleted(ref BlockParticipantCallbackInfo data)
+        {
+            OnLobbyCallback BlockParticipantCallback = data.ClientData as OnLobbyCallback;
+
+            if (data.ResultCode != Result.Success)
+            {
+                Debug.LogErrorFormat("Lobbies (OnRTCBlockParticipantCompleted): error code: {0}", data.ResultCode);
+                BlockParticipantCallback?.Invoke(data.ResultCode);
+                return;
+            }
+
+            Debug.LogFormat("Lobbies (OnRTCBlockParticipantCompleted): Blocked Participant successfully. Participant={0}, Room={1}, Blocked={2}", data.ParticipantId, data.RoomName, data.Blocked);
+
+            // Ensure this update is for our room
+            if (!CurrentLobby.RTCRoomName.Equals(data.RoomName, StringComparison.OrdinalIgnoreCase))
+            {
+                Debug.LogErrorFormat("Lobbies (OnRTCBlockParticipantCompleted): Incorrect Room! CurrentLobby.RTCRoomName={0} != data.RoomName", CurrentLobby.RTCRoomName, data.RoomName);
```

Comment:
Should we run the BlockParticipantCallback here?

It's a bit tricky, because the result code is success. So what do we invoke to say that this error happened? Maybe it's fine not to invoke, if this isn't the current room? I'm not sure.